hummingbot/market/kucoin/kucoin_api_order_book_data_source.py

```diff
@@ -91,23 +91,21 @@ def __init__(self, symbols: Optional[List[str]] = None):
             # Build the data frame.
             all_markets: pd.DataFrame = pd.DataFrame.from_records(data=market_data, index="symbol")
             # calculating the correct USDVolume by multiplying or diving with necessary USD pair
-            # split_symbol = all_markets.split('-')
-            if all_markets.symbol.startswith("USDT"):
-                all_markets.loc[:, "USDVolume"] = 1 / all_markets.volvalue
-            elif all_markets.symbol.endswith("USDT"):
-                all_markets.loc[:, "USDVolume"] = all_markets.volvalue
-            else:
-                try:
-                    quote_currency: str = symbol.split('-')[1]
-                    mul: str = quote_currency + "-USDT"
-                    div: str = "USDT-" + quote_currency
-                    if all_markets.loc[mul]:
-                        all_markets.loc[:, "USDVolume"] = all_markets.volvalue * all_markets.loc[mul].last
-                    else:
-                        all_markets.loc[:, "USDVolume"] = all_markets.volvalue / all_markets.loc[div].last
-                except Exception:
-                    self.logger().error(f"Unable to get USD Volume for {trading_pair}. ", exc_info=True)
-            all_markets.loc[:, "volume"] = all_markets.quoteVolume
+            for row in all_markets.itertuples():
+                product_name: str = row.Index
+                if product_name.startswith("USDT"):
+                    all_markets.loc[product_name, "USDVolume"] = 1 / float(row.volValue)
+                elif product_name.endswith("USDT"):
+                    all_markets.loc[product_name, "USDVolume"] = float(row.volValue)
+                else:
+                        quote_currency: str = product_name.split('-')[1]
+                        mul: str = quote_currency + "-USDT"
+                        div: str = "USDT-" + quote_currency
+                        if mul in all_markets.index:
+                            all_markets.loc[product_name, "USDVolume"] = float(row.volValue) * float(all_markets.loc[mul, "last"])
+                        elif div in all_markets.index:
+                            all_markets.loc[product_name, "USDVolume"] = float(row.volValue) / float(all_markets.loc[div, "last"])
+            all_markets.loc[:, "volume"] = all_markets.vol
```

Comment:
`get_active_exchange_markets()` is working now 👍 
However, it seems that it takes pretty long to derive the result. I would recommend modifying this sometime later. Maybe I will write something out myself when I have the chance

Reply:
Alright, I will. Thanks.

Reply:
@vic-en Can I get in touch with you regarding exchange connectors. ?

Reply:
@spmuppar, you can

Reply:
@vic-en what's your email id or discord id ?.
If you don't want share those details, please ping me on praneeth@pec.edu or my discord id "spmuppar#3706"